Changing where CodeEditor fields get useDarkMode value

kc13greiner opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/157203
Closes https://github.com/elastic/kibana/issues/158051
Closes https://github.com/elastic/kibana/issues/158049

With the addition of Per User Dark Mode, components can no longer rely on `uiSettings` as the source of truth for theme.

CodeEditor fields used to call `uiSettings` to determine if Dark Mode was enabled, which had been provided in each callers Kibana React Context.

The new source of truth for theme is the `CoreStart > ThemeServiceStart`.

Currently, not all callers of CodeEditor provide the `theme` service in their Kibana Context in a similar way and some callers don't provide it at all.

This PR updates CodeEditor to get theme values from the Kibana Context using a new `useKibanaTheme` react hook.

It also attempts audit the callers of CodeEditor to see if their Kibana Context contains the theme service at the top level (Where I could add theme to a caller's Context without major changes, I did. Some cases will require CodeOwner guidance).

The new `useKibanaTheme` react hook will throw a TypeError if theme isn't found in the top level of the Kibana Context, this will help with testing as the component will not render. I will remove this after testing so as not to introduce breaking changes.

## Testing

Please review files for which you are CODEOWNER.

I've attempted to tag all usages of `CodeEditor`/`CodeEditorFIeld` with a TODO comment with one of the following scenarios:

1) a note where theme was provided already/where I made changes to provide it in the appropriate context
2) I've asked for CODEOWNER guidance

For scenario 1, please pull and test that CodeEditor locally:

1. Enable Dark Mode from Edit User Profiles by clicking on the Profile Icon on the top right, and updating your profile.
2. Navigate to the CodeEditors in the plugins you own. If they render and display in Dark Mode - add a green check to the table below - and you're done!
3. If it is not rendering, please help me figure out where the theme service should be provided in the context.

For scenario 2, we will need to figure out where to make changes so your context is providing theme. Some of the more complex usages may need to addressed in separate issues.


## Tracking
| Team | Plugin | Theme in Context ? | Verified Working |
| - | - | - | - |
| apm-ui | apm | APM Storybook broken | ? |
| kibana-presentation | presentation_util | Yes. | Yes |
| response-ops | trigger_actions_ui | Yes | Yes |
| response-ops | stack_alerts | Yes | Yes |
| kibana-security | security  | Yes | Yes |
| security-defend-workflows | osquery | Yes | Yes |
| kibana-app-services | examples/expression_explorer | Yes | Yes |
| ml-ui | transform | Yes | Yes |
| ml-ui | ml | Yes | Yes |
| uptime | synthetics | Yes | Yes |
| kibana-gis | maps | Yes | Yes |
| kibana-gis | file_upload | Yes | Yes |
| platform-deployment-management | watcher | Yes | [AG] Yes |
| platform-deployment-management | snapshot_restore | Yes | [AG] Yes |
| platform-deployment-management | runtime_fields | Yes | [AG] Yes |
| platform-deployment-management | painless_lab | Yes | [AG] Yes |
| platform-deployment-management | ingest_pipelines | Yes | [AG] Yes |
| platform-deployment-management | index_management | Yes | [AG] Yes |
| platform-deployment-management | grokdebugger | Yes | [AG] Yes |
| platform-deployment-management | es_ui_shared | Yes | [AG] Yes |
| fleet | fleet | Yes | Yes |
| enterprise-search-frontend | enterprise_search | Yes | [AG] Yes |
| kibana-cloud-security-posture | cloud-security-posture | Yes | yes |
| sec-cloudnative-integrations | cloud_defend | Yes | Yes |
| kibana-visualizations/kibana-data-discovery | data | Yes | Yes |
| kibana-visualizations | examples/testing_embedded_lens | Yes | Yes |
| kibana-visualizations | vis_types | Yes | Yes |
| kibana-visualizations | vis_default_editor | Yes | Yes |
| kibana-visualizations | unified_search | Yes | Yes |
| kibana-visualizations | packages/kbn-text-based-editor | Yes | Yes |
| kibana-visualizatons | lens | Yes | Yes|
| kibana-core | saved_objects_management | Yes | Yes |
| kibana-presentation | inspector | Yes | Yes |
| kibana-presentation | canvas | Yes | Yes |
| kibana-data-discovery | discover | Yes | Yes |
| kibana-data-discovery | data_view_management | Yes | Yes |
| kibana-data-discovery | data_view_field_editor | Yes | Yes |
| appex-sharedux | advanced_settings | Yes | Yes |
| enterprise-search-frontend | serverless_search | Yes | [AG] Yes |
| - | - | - | - |

## Unit tests

Currently, many tests are failing since they are probably not providing `theme` in the context. Once CODEOWNERs have weighed in on CodeEditors usages that require discussion, I will update the accompanying tests.

## Release note
- Fixes theming of CodeEditors

<!--ONMERGE {"backportTargets":["8.9"]} ONMERGE-->

kc13greiner committed:
new theme hook

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

kc13greiner committed:
Adding TODOs to aid testers

kc13greiner committed:
Fixing theme in adv settings context and updating notes about snapshot_restore

kc13greiner committed:
Changing import/exports

kc13greiner committed:
Fixing CodeEditor to respond to Observable changes

kc13greiner commented:
@elasticmachine merge upstream

kibanamachine commented:
merge conflict between base and head

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

elasticmachine commented:
Pinging @elastic/kibana-security (Team:Security)

kc13greiner committed:
Adding tests and final comments

elasticmachine commented:
Pinging @elastic/apm-ui (Team:APM)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

TinaHeiligers reviewed (commented):
Code changes look fine in Saved Objects Management. I'll give official approval once tested.

@kc13greiner Hint: Deploying the PR to cloud will probably speed up reviews!

kc13greiner commented:
> @kc13greiner Hint: Deploying the PR to cloud will probably speed up reviews!

ahhh great call! Ill add the label - thanks!

TinaHeiligers reviewed (changes requested):
On the second pass-through, `theme` is [missing](https://github.com/elastic/kibana/blob/91fd3b69f2d07562011401af9289f258ac6ba248/src/plugins/saved_objects_management/public/management_section/object_view/saved_object_view.test.tsx#L78-L89) from `defaultProps` in `saved_object_view.test.tsx`. You'll need to [initialize](https://github.com/elastic/kibana/blob/main/src/plugins/saved_objects_management/public/management_section/object_view/saved_object_view.test.tsx#L57-L74) it in the `beforeEach` block with the `themeServiceMock`

stratoula commented on packages/kbn-text-based-editor/src/text_based_languages_editor.tsx:
```diff
@@ -515,7 +515,7 @@ export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
                       </EuiBadge>
                     )}
-                    <CodeEditor
+                    <CodeEditor // TODO Unsure how to proceed, theme would most likely be provided by all callers, need CodeOwners help
```
This seems to work fine now!

<img width="599" alt="image" src="https://github.com/elastic/kibana/assets/17003240/4765f642-4bf6-40fe-b3ad-ef5e5cd29e77">

stratoula commented on src/plugins/unified_search/public/filter_bar/filter_editor/filter_editor.tsx:
```diff
@@ -401,7 +401,7 @@ class FilterEditorComponent extends Component<FilterEditorProps, State> {
     return (
       <EuiFormRow fullWidth label={strings.getQueryDslLabel()} helpText={helpText}>
-        <CodeEditor
+        <CodeEditor // TODO CodeOwners will need to help with this
```
This seems to work fine now! 

<img width="1052" alt="image" src="https://github.com/elastic/kibana/assets/17003240/bd9cec2c-b1ed-4de2-8b3a-c55abe1dcfad">

stratoula commented on src/plugins/vis_default_editor/public/components/controls/raw_json.tsx:
```diff
@@ -79,7 +79,7 @@ function RawJsonParamEditor({
     >
       <>
-        <CodeEditor
+        <CodeEditor // TODO Unsure how to resolve context. Requires CODEOWNER input.
```
This also works fine now
<img width="652" alt="image" src="https://github.com/elastic/kibana/assets/17003240/29622ed2-9021-4618-bfe9-5e2a37908574">

stratoula commented on src/plugins/data/public/search/session/sessions_mgmt/components/actions/inspect_button.tsx:
```diff
@@ -33,7 +33,7 @@ const InspectFlyout = ({ uiSettings, searchSession }: InspectFlyoutProps) => {
     return (
       <Fragment>
-        <CodeEditor
+        <CodeEditor // TODO theme provided by src/plugins/data/public/search/search_service.ts
```
@elastic/kibana-data-discovery not sure which editor is that? Can you check if it is ok?

stratoula commented on src/plugins/vis_types/timelion/public/components/timelion_expression_input.tsx:
```diff
@@ -105,7 +105,7 @@ function TimelionExpressionInput({ value, setValue }: TimelionExpressionInputPro
       <div className="timExpressionInput__editor">
         <div className="timExpressionInput__absolute" data-test-subj="timelionCodeEditor">
-          <CodeEditor
+          <CodeEditor // TODO theme is provided in context by src/plugins/vis_types/timelion/public/timelion_vis_renderer.tsx
```
Timelion works fine now
<img width="748" alt="image" src="https://github.com/elastic/kibana/assets/17003240/e0c65cd1-c803-47be-85cd-29453bd02b73">

stratoula commented on src/plugins/vis_types/timeseries/public/application/components/markdown_editor.js:
```diff
@@ -122,7 +122,7 @@ export class MarkdownEditor extends Component {
       <div className="tvbMarkdownEditor">
         <div className="tvbMarkdownEditor__editor">
-          <CodeEditor
+          <CodeEditor // TODO theme is provided in context by src/plugins/vis_types/timeseries/public/application/components/vis_editor.tsx
```
markdown seems correct
<img width="1106" alt="image" src="https://github.com/elastic/kibana/assets/17003240/0074e184-438b-4bc7-929d-fc94f01c1155">

stratoula commented on src/plugins/vis_types/timeseries/public/application/components/panel_config/markdown.tsx:
```diff
@@ -257,7 +257,7 @@ export class MarkdownPanelConfig extends Component<
             </EuiTitle>
             <EuiSpacer size="s" />
-            <CodeEditor
+            <CodeEditor // TODO theme is provided in context by src/plugins/vis_types/timeseries/public/application/components/vis_editor.tsx
```
Same 
<img width="1139" alt="image" src="https://github.com/elastic/kibana/assets/17003240/1059b286-eb1e-4751-bf18-52b28578ad98">

stratoula commented on src/plugins/vis_types/vega/public/components/vega_vis_editor.tsx:
```diff
@@ -101,7 +101,7 @@ function VegaVisEditor({ stateParams, setValue }: VisEditorOptionsProps<VisParam
         <VegaActionsMenu formatHJson={formatHJson} formatJson={formatJson} />
       </div>
-      <CodeEditor
+      <CodeEditor // TODO theme needs to be provided, need CodeOwner help
```
Vega seems fine
<img width="701" alt="image" src="https://github.com/elastic/kibana/assets/17003240/5512ca59-f011-4ee4-a525-1ed34b527b3a">

stratoula commented on src/plugins/vis_types/vega/public/vega_inspector/components/spec_viewer.tsx:
```diff
@@ -71,7 +71,7 @@ export const SpecViewer = ({ vegaAdapter, ...rest }: SpecViewerProps) => {
       </EuiFlexItem>
       <EuiFlexItem grow={true}>
-        <CodeEditor
+        <CodeEditor // TODO theme needs to be provided from src/plugins/vis_types/vega/public/vega_inspector/vega_inspector.tsx, need Code Owner help
```
Spec viewer works as expected
<img width="1098" alt="image" src="https://github.com/elastic/kibana/assets/17003240/f03fe903-f4aa-4002-9bf7-4bfcd6c21448">

stratoula commented on x-pack/plugins/lens/public/datasources/form_based/operations/definitions/formula/editor/formula_editor.tsx:
```diff
@@ -779,7 +779,7 @@ export function FormulaEditor({
 
             <div className="lnsFormula__editorContent">
-              <CodeEditor
+              <CodeEditor // TODO theme provided in context by x-pack/plugins/lens/public/app_plugin/mounter.tsx
```
Lens formula is broken , I see this error in the console

<img width="837" alt="image" src="https://github.com/elastic/kibana/assets/17003240/16d30983-e488-48d0-88bc-60653db9499b">

kc13greiner replied:
Ah thank you! Fixed in  https://github.com/elastic/kibana/pull/159638/commits/0ff4ad74db0e8711d4640244f453a9b9f9c4b347

stratoula reviewed (changes requested):
Except from Lens formula which is broken, everything else works as expected.

@kc13greiner I see that you have added comments about needing help form us but everything seems to work fine except from Lens.

kc13greiner committed:
Code Review Feedback from Tina H.

kc13greiner commented:
> On the second pass-through, `theme` is [missing](https://github.com/elastic/kibana/blob/91fd3b69f2d07562011401af9289f258ac6ba248/src/plugins/saved_objects_management/public/management_section/object_view/saved_object_view.test.tsx#L78-L89) from `defaultProps` in `saved_object_view.test.tsx`. You'll need to [initialize](https://github.com/elastic/kibana/blob/main/src/plugins/saved_objects_management/public/management_section/object_view/saved_object_view.test.tsx#L57-L74) it in the `beforeEach` block with the `themeServiceMock`

@TinaHeiligers great catch! Fixed in https://github.com/elastic/kibana/pull/159638/commits/5ebd0b2843696e903b893d46618f7b08c86a4441

kc13greiner committed:
CodeReview feed back from Stratoula K.

TinaHeiligers reviewed (approved):
Works like a charm!
![Screenshot 2023-06-21 at 13 18 44](https://github.com/elastic/kibana/assets/11909450/9c4a9617-91c1-4aee-87f0-221e9c935f31)

stratoula commented:
@kc13greiner thanx, Lens formula also works fine now. Can we remove the comments from the codebase as everything works as expected?

sqren commented on x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/settings_form/form_row_setting.tsx:
```diff
@@ -125,7 +125,7 @@ export function FormRowSetting({ row, value, onChange, isDisabled }: Props) {
       return (
         <FixedHeightDiv>
-          <CodeEditor
+          <CodeEditor // TODO Unable to find context that provides theme. Need CODEOWNER Input
```
@ogupte Does this not have access to the Kibana context supplied here:
https://github.com/elastic/kibana/blob/8861c27f44e70e8853e508887ee599e8f4a4341e/x-pack/plugins/apm/public/components/routing/app_root/index.tsx#L69

(I'm guessing it might be embedded outside our normal application)

kc13greiner replied:
@sqren @ogupte Has anyone been able to verify if this editor is working?

kc13greiner replied:
Last one here - let me know if I can be of assistance to get this verified!

ogupte replied:
This is from a component that is rendered in the Fleet UI (via fleet.registerExtension (https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/public/plugin.ts#L407-L439). It appears to be getting the correct theming with manual testing:
![Screenshot 2023-07-11 at 00 04 40](https://github.com/elastic/kibana/assets/1967266/3e77d4e9-8eaa-4aed-93d7-0f9c8887bce3)

kc13greiner replied:
Excellent - thank you for verifying!

ogupte replied:
I just tested the user profile only setting, and it appears the CodeEditor picks it up just fine, but the rest of the Fleet UI doesn't appear to use the correct theme:
![Screenshot 2023-07-11 at 09 45 28](https://github.com/elastic/kibana/assets/1967266/9d89a2bc-5d1e-4f3e-b48a-415ad1854736)

kc13greiner replied:
This PR is only to address CodeEditors.

I created an [issue](https://github.com/elastic/kibana/issues/161135) for the Fleet Dark Mode UI issues. 

Here is the Dark Mode Issues meta [issue](https://github.com/elastic/kibana/issues/158201)

sqren commented on x-pack/plugins/apm/public/components/app/service_map/__stories__/cytoscape_example_data.stories.tsx:
```diff
@@ -105,7 +105,7 @@ export const GenerateMap: Story<{}> = () => {
 
       {json && (
-        <CodeEditor
+        <CodeEditor // TODO Unable to find context that provides theme. Need CODEOWNER Input
```
I think you need to add the necessary context when invoking `MockApmPluginStorybook`:

https://github.com/elastic/kibana/blob/ca425e899344a6381e05113c93f861dd44201d86/x-pack/plugins/apm/public/components/app/service_map/__stories__/cytoscape_example_data.stories.tsx#L49

Something like this:
```
<MockApmPluginStorybook apmContext={{ foo: 'bar' }}>
```

This is passed on to the Kibana Context Provider here:
https://github.com/elastic/kibana/blob/8861c27f44e70e8853e508887ee599e8f4a4341e/x-pack/plugins/apm/public/context/apm_plugin/mock_apm_plugin_storybook.tsx#L136-L140

kc13greiner replied:
Sorry, I'm struggling to figure this out; can you link me to where the `GenerateMap` and `MapFromJson` story functions are being used?

sqren replied:
It's used for storybook: https://ci-artifacts.kibana.dev/storybooks/main/eac8f4ce25737e7f2dab5d9d131e2a414c4fa47b/apm/index.html?path=/story/app-servicemap-example-data--map-from-json

You can run it locally with `yarn storybook apm`

kc13greiner replied:
@sqren based on the Slack discussion it seems these CodeEditors (in Storybook) might have other issues going on. Would it be alright to move forward with this PR and address them at a later date?

sqren replied:
Yes, fine to ignore the storybook failures for now. Sorry for being slow to follow up on this.

kc13greiner replied:
No worries! Thanks for the update.

sphilipse commented on x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/sync_rules/advanced_sync_rules.tsx:
```diff
@@ -40,7 +40,7 @@ export const AdvancedSyncRules: React.FC = () => {
       fullWidth
     >
-      <CodeEditor
+      <CodeEditor // TODO theme is provided in context by x-pack/plugins/enterprise_search/public/applications/index.tsx
```
What is the actual TODO here? It's unclear to me what action needs to be taken.

kc13greiner replied:
Heya @sphilipse ! The `CodeEditor` marked with the TODO should exist somewhere within a plugin owned by your team (Hopefully I got it right!). Being unfamiliar with where all the `CodeEditor` fields exist in the app, I'm asking the owners to help me verify that their `CodeEditor`s are displaying as expected.

Please enable Dark Mode from Edit User Profiles by clicking on the Profile Icon on the top right, and updating your profile.

Navigate to the CodeEditor mentioned above in the app. If they render and display in Dark Mode - add a value to the `Verified Working` table in the PR description  - and you're done!

If it is not rendering, please help me figure out where the theme service should be provided in the context.

alisonelizabeth commented on x-pack/plugins/snapshot_restore/__jest__/client_integration/helpers/setup_environment.tsx:
```diff
@@ -61,7 +61,7 @@ const appDependencies = {
 };
-
+// TODO need to provide theme in x-pack/plugins/snapshot_restore/public/application/app_context.tsx and mock here
 const kibanaContextDependencies = {
```
```suggestion
const kibanaContextDependencies = {
  uiSettings: core.uiSettings,
  settings: core.settings,
  theme: core.theme,
};
```

alisonelizabeth reviewed (commented):
Thanks for working on this @kc13greiner! Verified the Management team changes and updated the table. I left a couple comments for the `snapshot_restore` plugin. I believe everything is working as expected, only the jest test setup needs to be updated.

kc13greiner committed:
CodeReview feedback for Stratoula K. - removing TODOs

kc13greiner commented:
> @kc13greiner thanx, Lens formula also works fine now. Can we remove the comments from the codebase as everything works as expected?

@stratoula Thanks again for the quick feedback! I've removed the TODOs for the CodeEditors you've approved: https://github.com/elastic/kibana/pull/159638/commits/79b0be071aa40cffb581a3c080aca6e9e714518c

I've left the TODO on the line referenced [here](https://github.com/elastic/kibana/pull/159638#discussion_r1236590874)

kc13greiner commented on x-pack/plugins/snapshot_restore/public/application/app_context.tsx:
```diff
@@ -37,6 +37,12 @@ export const AppContextProvider = ({
   children: React.ReactNode;
 }) => {
+  /*
+   * TODO theme needs to be provided here for
```
Ah perfect! Sorry for the confusion - as long as the `CodeEditor`s are displaying, no changes should be required!

kc13greiner commented on src/plugins/es_ui_shared/public/components/json_editor/json_editor.tsx:
```diff
@@ -83,7 +83,7 @@ function JsonEditorComp<T extends object = { [key: string]: any }>({
       fullWidth
     >
-      <CodeEditor
+      <CodeEditor // TODO JsonEditor is used many places. Need CodeOwners help to verify. One Example: theme provided in x-pack/plugins/index_management/public/application/index.tsx
```
Excellent - thank you!

kc13greiner committed:
Update x-pack/plugins/snapshot_restore/__jest__/client_integration/helpers/setup_environment.tsx

Co-authored-by: Alison Goryachev <alisonmllr20@gmail.com>

kc13greiner committed:
CodeReview feedback from Alison G. - removing TODOs and fixing test env data

kc13greiner commented:
> Thanks for working on this @kc13greiner! Verified the Management team changes and updated the table. I left a couple comments for the `snapshot_restore` plugin. I believe everything is working as expected, only the jest test setup needs to be updated.

@alisonelizabeth Thanks for the quick feedback on this! I've updated the test and removed the TODOs https://github.com/elastic/kibana/pull/159638/commits/d3d6458055151bac60070ca678d36c7a4d28ed1a

alisonelizabeth reviewed (approved):
Latest Deployment Management changes LGTM

opauloh commented on x-pack/plugins/cloud_security_posture/public/pages/configurations/findings_flyout/json_tab.tsx:
```diff
@@ -14,7 +14,7 @@ const offsetHeight = 120;
 export const JsonTab = ({ data }: { data: CspFinding }) => (
   <div style={{ position: 'absolute', inset: 0, top: offsetHeight }}>
-    <CodeEditor
+    <CodeEditor // TODO theme provided in context by x-pack/plugins/cloud_security_posture/public/plugin.tsx
```
tested in CI deploy and it looks fine

kc13greiner replied:
Excellent!

opauloh commented on x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities_finding_flyout/vulnerability_json_tab.tsx:
```diff
@@ -21,7 +21,7 @@ export const VulnerabilityJsonTab = ({ vulnerabilityRecord }: VulnerabilityJsonT
       data-test-subj={JSON_TAB_VULNERABILITY_FLYOUT}
     >
-      <CodeEditor
+      <CodeEditor // TODO theme provided in context by x-pack/plugins/cloud_security_posture/public/plugin.tsx
```
it works fine!

kc13greiner replied:
Thanks for the quick turn around on this! Ill remove the TODOs now!

opauloh reviewed (approved):
cloud-security-posture LGTM

kc13greiner committed:
CodeReview feedback from Paulo H.

kc13greiner commented:
Thanks again, @opauloh ! TODOs removed https://github.com/elastic/kibana/pull/159638/commits/e29211bfe59ca1b2de4071c4dd360048dc76ae79

ThomThomson commented on src/plugins/inspector/public/views/requests/components/details/req_code_viewer.tsx:
```diff
@@ -133,7 +133,7 @@ export const RequestCodeViewer = ({ indexPattern, json }: RequestCodeViewerProps
       </EuiFlexItem>
       <EuiFlexItem grow={true} data-test-subj="inspectorRequestCodeViewerContainer">
-        <CodeEditor
+        <CodeEditor // TODO theme provided in context in src/plugins/inspector/public/ui/inspector_panel.tsx
```
Theme looks good here:

<img width="1167" alt="Screen Shot 2023-06-22 at 6 10 20 PM" src="https://github.com/elastic/kibana/assets/14276393/476208ed-9019-4ce8-854f-1df6237d6eed">

kc13greiner replied:
Nice!

ThomThomson commented on x-pack/plugins/canvas/canvas_plugin_src/uis/datasources/essql.js:
```diff
@@ -77,7 +77,7 @@ class EssqlDatasource extends PureComponent {
         }
       >
-        <CodeEditorField
+        <CodeEditorField // TODO theme appears to be provided in x-pack/plugins/canvas/public/application.tsx
```
Dark theme looks good here:

The white on the left side is the actual contents of the Canvas workpad which is not intended to update.

<img width="418" alt="Screen Shot 2023-06-22 at 6 12 30 PM" src="https://github.com/elastic/kibana/assets/14276393/287eb9de-1205-4a87-9675-abe148996ae9">

kc13greiner replied:
Ah ok - thanks!

ThomThomson commented on x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/editor.tsx:
```diff
@@ -53,7 +53,7 @@ const EditorArg: FC<EditorArgProps> = ({ argValue, typeInstance, onValueChange,
   return (
     <EuiFormRow display="rowCompressed" data-test-subj="canvasCodeEditorField">
-      <CodeEditorField
+      <CodeEditorField // TODO theme appears to be provided in x-pack/plugins/canvas/public/application.tsx
```
Looks good!
<img width="337" alt="Screen Shot 2023-06-22 at 6 14 12 PM" src="https://github.com/elastic/kibana/assets/14276393/0f27285f-76f5-4750-a372-6a6588b858e6">

kc13greiner replied:
++

ThomThomson commented on x-pack/plugins/file_upload/public/components/import_complete_view.tsx:
```diff
@@ -76,7 +76,7 @@ export class ImportCompleteView extends Component<Props, {}> {
         </EuiFlexGroup>
         <div style={{ height: '200px' }}>
-          <CodeEditor
+          <CodeEditor // TODO need to provide theme in context above. Need CodeOwners help.
```
@nreese I'm not exactly sure how to get to this screen. I think I might need to successfully upload some features first? Would you mind verifying this one and the `tile_request_tab`?

nreese replied:
Expose getTheme method in https://github.com/elastic/kibana/blob/main/x-pack/plugins/file_upload/public/kibana_services.ts and then use the getter function to get theme in this component

kc13greiner replied:
@nreese On which screen can I find this CodeEditor? Would you mind helping me navigate again 😅

nreese replied:
* Download world countries geojson at https://vector.maps.elastic.co/files/world_countries_v7.geo.json?elastic_tile_service_tos=agree&my_app_name=ems-landing-page&my_app_version=8.8.1&license=643c1faf-80fc-4ab0-9323-4d9bd11f4bbc and save into a file
* Create new map
* Click "Add layer"
* Select "Upload"
* Select world countries geo json file
* Import file.
* Results displayed in code editor.

kc13greiner replied:
Thanks @nreese ! I added `theme` to `kibana_services`and it's working now. See https://github.com/elastic/kibana/pull/159638/commits/abcc1700f68ca99c53d324e681bed86aa5691064
<img width="458" alt="Screenshot 2023-06-28 at 3 53 16 PM" src="https://github.com/elastic/kibana/assets/21210601/83266c69-f6d0-4f91-98f8-e82107f169ad">

ThomThomson reviewed (approved):
Presentation team code in Canvas and Inspector is looking good! I changed my user to dark theme and verified locally that all of the code editors were receiving the correct theme and that there were no errors.

I wasn't able to figure out how to see the Maps  changes, @elastic/kibana-gis will likely need to cover those files.

stratoula reviewed (approved):
Visualizations team changes LGTM! @elastic/kibana-data-discovery team can you check the search sessions?

kc13greiner committed:
Code Review feedback from Devon T. - removing TODOs

kc13greiner commented:
Thanks @ThomThomson! I've removed the TODOs from the `CodeEditor`s you verified https://github.com/elastic/kibana/pull/159638/commits/9f48c1016ff3408616553dd2a4c80845161e8296

kc13greiner commented on src/plugins/data_view_field_editor/public/components/field_editor/form_fields/script_field.tsx:
```diff
@@ -259,7 +259,7 @@ const ScriptFieldComponent = ({ existingConcreteFields, links, placeholder }: Pr
               fullWidth
             >
-              <CodeEditor
+              <CodeEditor // TODO theme provided in src/plugins/data_view_field_editor/public/open_editor.tsx
```
Excellent!

kc13greiner commented on src/plugins/data_view_management/public/components/field_editor/field_editor.tsx:
```diff
@@ -601,7 +601,7 @@ export class FieldEditor extends PureComponent<FieldEdiorProps, FieldEditorState
           error={isInvalid ? errorMsg : null}
         >
-          <CodeEditor
+          <CodeEditor // TODO theme provided in src/plugins/data_view_management/public/management_app/mount_management_section.tsx
```
++

kc13greiner commented on src/plugins/data/public/search/session/sessions_mgmt/components/actions/inspect_button.tsx:
```diff
@@ -33,7 +33,7 @@ const InspectFlyout = ({ uiSettings, searchSession }: InspectFlyoutProps) => {
     return (
       <Fragment>
-        <CodeEditor
+        <CodeEditor // TODO theme provided by src/plugins/data/public/search/search_service.ts
```
Awesome, thank you for the quick turnaround! This PR will definitely be around for a while, we can definitely wait to merge until after #160358. I am targeting 8.9 as this PR is a bugfix, will you be able to get your PR merged into 8.9?

jughosta replied:
Yes, I think I will be able to backport it.

jughosta replied:
The mention PR is merged. Let's merge `main` in and I review again.

kc13greiner replied:
Sorry for the delay - main is merged in now

jughosta replied:
Thanks! Works now. The TODO can be removed:

<img width="500" alt="Screenshot 2023-06-28 at 10 14 05" src="https://github.com/elastic/kibana/assets/1415710/d99c05ad-c924-4615-b898-a0aac295ccf9">

shahzad31 commented on x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/fields/code_editor.tsx:
```diff
@@ -57,7 +57,7 @@ export const CodeEditor = ({
           data-test-subj="codeEditorContainer"
         >
-          <MonacoCodeEditor
+          <MonacoCodeEditor // TODO theme provided in context by x-pack/plugins/synthetics/public/apps/synthetics/synthetics_app.tsx
```
Should we create a follow up issue for this?

kc13greiner replied:
Is it failing to render? Based on my search (with limited understanding of that plugin 😅 ), it looked like `theme` was being provided.

If it is throwing an error, we can fix it here, or, if it is a bigger change we can create a new issue

kc13greiner replied:
Heya @shahzad31! Can you verify these `CodeEditor`s aren't rendering? Or would you rather create another issue to address this later?

shahzad31 replied:
it seems to be working 
<img width="1390" alt="image" src="https://github.com/elastic/kibana/assets/3505601/767e45b9-cad5-46ec-b65a-d952dc8850d3">

rshen91 commented on src/plugins/kibana_react/public/code_editor/code_editor.tsx:
```diff
@@ -357,6 +357,15 @@ export const CodeEditor: React.FC<Props> = ({
     ]
   );
 
+  useEffect(() => {
```
Is it possible to migrate these changes to the shared-ux package in packages/shared-ux/code_editor/impl/code_editor.tsx. AFAIK we are hoping to de-duplicate the code from kibana react to the shared ux component (that was originally taken from this component).

kc13greiner replied:
Heya @rshen91! It would great to consolidate and migrate all of these `CodeEditor`s to the new package, but I would like to do it in a future PR. I want to avoid having all my reviewers/testers having to re-test/re-review as this has to ship with 8.9.

Would it be beneficial if I copied the new code into the shared-ux `CodeEditor`  in this PR or should we wait until the consolidation?

rshen91 replied:
all good to do in a future PR!

kc13greiner replied:
++ Thank you!

rshen91 commented on src/plugins/kibana_react/public/theme/use_theme.test.tsx:
```diff
@@ -0,0 +1,56 @@
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
```
can we also have this file and use_theme moved under packages/shared-ux/code_editor/impl? Thanks!

nchaulet reviewed (approved):
Fleet changes LGTM 🚀

rshen91 reviewed (changes requested):
These changes are awesome! We have an issue https://github.com/elastic/kibana/issues/159719 where we are hoping to not have to maintain the kibana react AND the shared ux code editor. These changes are awesome - would it be possible to have these changes brought into the shared ux code editor package please?

nreese commented on x-pack/plugins/maps/public/inspector/vector_tile_adapter/components/tile_request_tab.tsx:
```diff
@@ -90,7 +90,7 @@ export function TileRequestTab(props: Props) {
         </EuiFlexItem>
         <EuiFlexItem grow={true}>
-          <CodeEditor
+          <CodeEditor // TODO Unable to find context being provided. Need help from CodeOwners
```
You can import a getter function to access theme service in maps plugin `import { getTheme } from '../../../kibana_services';`

kc13greiner replied:
I'm having trouble navigating to this CodeEditor, can you help me figure out where it is?

nreese replied:
* install sample web logs data set
* create new map
* Set time range to last 24 hours
* Click "Add layer"
* Select "Documents"
* Select sample logs data view
* You should see dots on a map now
    <img width="500" alt="Screen Shot 2023-06-26 at 1 26 12 PM" src="https://github.com/elastic/kibana/assets/373691/5d0b8fba-594a-4e03-98b2-e4eacd3c9397">
* Click "Inspect". The inspector will open for "Vector tiles" and you can see the request body in the code editor
    <img width="500" alt="Screen Shot 2023-06-26 at 1 27 22 PM" src="https://github.com/elastic/kibana/assets/373691/5b129f1c-fbb3-42a6-97a9-731a34cd5dbc">

kc13greiner replied:
++ thank you!! That `CodeEditor` looks like it has the theme being provided (I am not super familiar with that plugin):
<img width="1274" alt="Screenshot 2023-06-26 at 3 33 20 PM" src="https://github.com/elastic/kibana/assets/21210601/cbf7b267-f93c-47b7-9ebf-449dedfe625c">

nreese commented on src/plugins/presentation_util/public/components/expression_input/expression_input.tsx:
```diff
@@ -75,7 +75,7 @@ export const ExpressionInput = (props: ExpressionInputProps) => {
   return (
     <div style={{ height, ...style }} {...{ rest }}>
-      <CodeEditor
+      <CodeEditor // TODO CodeOwners will need to help with this one
```
Expose getTheme method in https://github.com/elastic/kibana/blob/main/x-pack/plugins/file_upload/public/kibana_services.ts and then use the getter function to get theme in this component

kc13greiner committed:
Code Review feedback from Sander P. - removing TODO

kc13greiner commented:
> Enterprise Search Frontend and Serverless Search changes LGTM

@sphilipse Thank you! Removing the TODO https://github.com/elastic/kibana/pull/159638/commits/4aa4657a4e99d89267d2a6ba5193afdcc0f46396

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

jughosta commented on src/plugins/data_view_field_editor/public/components/field_editor/form_fields/script_field.tsx:
```diff
@@ -259,7 +259,7 @@ const ScriptFieldComponent = ({ existingConcreteFields, links, placeholder }: Pr
               fullWidth
             >
-              <CodeEditor
+              <CodeEditor // TODO theme provided in src/plugins/data_view_field_editor/public/open_editor.tsx
```
Works now. The TODO can be removed:

<img width="500" alt="Screenshot 2023-06-28 at 10 15 02" src="https://github.com/elastic/kibana/assets/1415710/246117bc-c58f-4f0a-a038-e00971316405">

jughosta commented on src/plugins/data_view_management/public/components/field_editor/field_editor.tsx:
```diff
@@ -601,7 +601,7 @@ export class FieldEditor extends PureComponent<FieldEdiorProps, FieldEditorState
           error={isInvalid ? errorMsg : null}
         >
-          <CodeEditor
+          <CodeEditor // TODO theme provided in src/plugins/data_view_management/public/management_app/mount_management_section.tsx
```
Works now. The TODO can be removed:

<img width="500" alt="Screenshot 2023-06-28 at 10 20 18" src="https://github.com/elastic/kibana/assets/1415710/2b04065e-9bb0-432d-9e2c-136b5de9dde9">

jughosta commented on src/plugins/discover/public/build_services.ts:
```diff
@@ -63,7 +63,7 @@ import { DiscoverAppLocator } from '../common';
   referrer: string;
 }
-
+// TODO theme needs to be provided with the theme$ observable, let code owners decide
```
`theme` was added to `DiscoverServices`. The TODO can be removed.

jughosta commented on src/plugins/discover/public/components/json_code_editor/json_code_editor_common.tsx:
```diff
@@ -42,7 +42,7 @@ export const JsonCodeEditorCommon = ({
   }
   const codeEditor = (
-    <CodeEditor
+    <CodeEditor // TODO theme needs to be provided in the top level context, in src/plugins/discover/public/build_services.ts, theme is not providing a theme$ observable
```
JSON views work on Discover, including Surrounding documents and Single document pages. The TODO can be removed.

<img width="500" alt="Screenshot 2023-06-28 at 10 14 18" src="https://github.com/elastic/kibana/assets/1415710/079f902e-44f3-43ff-a28a-f3b03de2cc0d">
<img width="500" alt="Screenshot 2023-06-28 at 10 13 20" src="https://github.com/elastic/kibana/assets/1415710/a5858a30-e580-4342-abfa-6a658357c7d1">

jughosta reviewed (approved):
Works correctly on Data Discovery pages. LGTM 👍

kc13greiner committed:
CodeReview feedback from Julia R. - Removing TODOs

kc13greiner commented:
Thanks @jughosta! Removing TODOS here: https://github.com/elastic/kibana/pull/159638/commits/88fa75781951d72f81a69868b29a2a4cced2a9c7

kc13greiner committed:
Code Review feedback from Nathan R. - adding theme to the file_upload kibana_services and using that to provide them in the context above the CodeEditor in import_complete_view

kc13greiner commented:
> Enterprise Search Frontend and Serverless Search changes LGTM

Heya @sphilipse! Would you mind adding your approval as well?

nreese commented on x-pack/plugins/file_upload/public/components/import_complete_view.tsx:
```diff
@@ -221,8 +221,9 @@ export class ImportCompleteView extends Component<Props, {}> {
+    const theme = getTheme();
     return (
-      <KibanaContextProvider services={services}>
+      <KibanaContextProvider services={{ ...services, theme }}>
```
Thanks. Nit, could you add theme to services object creation above instead of here

so change 
```
const services = {
  uiSettings: getUiSettings(),
  settings: getSettings(),
};
```

to 

```
const services = {
  uiSettings: getUiSettings(),
  settings: getSettings(),
  theme: getTheme(),
};
```

kc13greiner replied:
++ That is cleaner, done! https://github.com/elastic/kibana/pull/159638/commits/21c6350b5e1aeb466a07c4c4fee18848321883f3

kc13greiner committed:
Code Review feedback from Nathan R.

nreese reviewed (approved):
kibana-gis changes LGTM

kc13greiner committed:
Code Review feedback from Nathan R. - removing TODOs

kc13greiner commented:
@nreese Thank for helping get that figured out - removing TODOs here https://github.com/elastic/kibana/pull/159638/commits/d070e98efc203788c1942800ab69fcc0b9d6a89c

kc13greiner committed:
Code Review feedback from Rachel S. - removing TODOs

kc13greiner commented:
Thanks @rshen91! removing TODO here https://github.com/elastic/kibana/pull/159638/commits/701dff486cdcdb555fb8e3b8624c8402e5630062

kc13greiner committed:
Code Review feedback from Tina H. - Removing TODO

kc13greiner commented:
@TinaHeiligers Thanks again! Removing TODO https://github.com/elastic/kibana/pull/159638/commits/0bf462338fffdd5d719b3ea8ac0db2ccc31b49cd

kc13greiner committed:
Removing todo - context was provided

kc13greiner committed:
Code Review feedback from Karl G.  - Removing TODO

kc13greiner committed:
Code Review feedback Sander P

kc13greiner commented:
@sphilipse removing TODOs here https://github.com/elastic/kibana/pull/159638/commits/65029579e4be8e52d6efe227bc8742b76b796a5e

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner committed:
Code Review from Nicolas C. - removing TODOs

kc13greiner committed:
Code Review from Alison G. - Removing TODOs

kc13greiner committed:
Code Review from Alison G. - removing TODOs

kc13greiner committed:
Code Review from Alison G - Removing TODOs

darnautov committed:
fix MLJobEditor theme

darnautov commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/create_analytics_advanced_editor/create_analytics_advanced_editor.tsx:
```diff
@@ -139,7 +139,7 @@ export const CreateAnalyticsAdvancedEditor: FC<CreateAnalyticsFormProps> = (prop
         style={{ maxWidth: '100%' }}
       >
-        <CodeEditor
+        <CodeEditor // TODO theme provided in context by x-pack/plugins/ml/public/application/app.tsx
```
```suggestion
        <CodeEditor
```
works correctly

darnautov commented on x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/runtime_mappings/runtime_mappings_editor.tsx:
```diff
@@ -33,7 +33,7 @@ export const RuntimeMappingsEditor: FC<Props> = memo(
     return (
       <div data-test-subj="mlDataFrameAnalyticsAdvancedRuntimeMappingsEditor">
-        <CodeEditor
+        <CodeEditor // TODO theme provided in context by x-pack/plugins/ml/public/application/app.tsx
```
```suggestion
        <CodeEditor
```
works correctly

darnautov commented on x-pack/plugins/transform/public/app/sections/create_transform/components/advanced_pivot_editor/advanced_pivot_editor.tsx:
```diff
@@ -29,7 +29,7 @@ export const AdvancedPivotEditor: FC<StepDefineFormHook['advancedPivotEditor']>
         data-test-subj="transformAdvancedPivotEditor"
       >
-        <CodeEditor
+        <CodeEditor // TODO theme is provided in context by x-pack/plugins/transform/public/app/app.tsx
```
```suggestion
        <CodeEditor
```
👍🏻

darnautov commented on x-pack/plugins/transform/public/app/sections/create_transform/components/advanced_runtime_mappings_editor/advanced_runtime_mappings_editor.tsx:
```diff
@@ -26,7 +26,7 @@ export const AdvancedRuntimeMappingsEditor: FC<StepDefineFormHook['runtimeMappin
     return (
       <div data-test-subj="transformAdvancedRuntimeMappingsEditor">
-        <CodeEditor
+        <CodeEditor // TODO theme is provided in context by x-pack/plugins/transform/public/app/app.tsx
```
```suggestion
        <CodeEditor
```
👍🏻

darnautov commented on x-pack/plugins/transform/public/app/sections/create_transform/components/advanced_source_editor/advanced_source_editor.tsx:
```diff
@@ -24,7 +24,7 @@ export const AdvancedSourceEditor: FC<StepDefineFormHook> = ({
   return (
     <div data-test-subj="transformAdvancedRuntimeMappingsEditor">
-      <CodeEditor
+      <CodeEditor // TODO theme is provided in context by x-pack/plugins/transform/public/app/app.tsx
```
```suggestion
      <CodeEditor
```
👍🏻

darnautov commented on x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/filter_agg/components/editor_form.tsx:
```diff
@@ -19,7 +19,7 @@ export const FilterEditorForm: FilterAggConfigEditor['aggTypeConfig']['FilterAgg
     <>
       <EuiSpacer size="m" />
-      <CodeEditor
+      <CodeEditor // TODO theme is provided in context by x-pack/plugins/transform/public/app/app.tsx
```
```suggestion
      <CodeEditor
```
👍🏻

darnautov reviewed (approved):
ML and Transform tested, pushed a [fix](https://github.com/elastic/kibana/pull/159638/commits/ac527b0645d525a0ce489f9c57540430ba55b996) for one component.

kc13greiner committed:
Update x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/create_analytics_advanced_editor/create_analytics_advanced_editor.tsx

Co-authored-by: Dima Arnautov <arnautov.dima@gmail.com>

kc13greiner committed:
Update x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/runtime_mappings/runtime_mappings_editor.tsx

Co-authored-by: Dima Arnautov <arnautov.dima@gmail.com>

kc13greiner committed:
Update x-pack/plugins/transform/public/app/sections/create_transform/components/advanced_pivot_editor/advanced_pivot_editor.tsx

Co-authored-by: Dima Arnautov <arnautov.dima@gmail.com>

kc13greiner committed:
Update x-pack/plugins/transform/public/app/sections/create_transform/components/advanced_runtime_mappings_editor/advanced_runtime_mappings_editor.tsx

Co-authored-by: Dima Arnautov <arnautov.dima@gmail.com>

kc13greiner committed:
Update x-pack/plugins/transform/public/app/sections/create_transform/components/advanced_source_editor/advanced_source_editor.tsx

Co-authored-by: Dima Arnautov <arnautov.dima@gmail.com>

kc13greiner committed:
Update x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/filter_agg/components/editor_form.tsx

Co-authored-by: Dima Arnautov <arnautov.dima@gmail.com>

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner committed:
Updating jest tests

kc13greiner committed:
Fixing URL Drilldown CodeEditor and Functional Test

kc13greiner committed:
Removing debug statement

kc13greiner committed:
Passing CoreStart to props for expression editor

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner committed:
Updating props and unit test

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

kc13greiner committed:
Adding theme to fleet kb context

patrykkopycinski commented on x-pack/plugins/osquery/public/editor/index.tsx:
```diff
@@ -77,7 +77,7 @@ const OsqueryEditorComponent: React.FC<OsqueryEditorProps> = ({
 
   return (
-    <CodeEditor
+    <CodeEditor // TODO theme provided in context by x-pack/plugins/osquery/public/application.tsx
```
it's done here https://github.com/elastic/kibana/blob/main/x-pack/plugins/osquery/public/application.tsx#L58

kc13greiner replied:
Ok perfect! Ill remove the TODO and tag you - thanks for verifying!

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner committed:
Changing imports to remove circular deps

cnasikas commented on x-pack/plugins/triggers_actions_ui/public/application/components/json_editor_with_message_variables.tsx:
```diff
@@ -173,7 +173,7 @@ export const JsonEditorWithMessageVariables: React.FunctionComponent<Props> = ({
       <>
         {renderErrorMessage()}
-        <CodeEditor
+        <CodeEditor // TODO theme provided in context by x-pack/plugins/triggers_actions_ui/public/application/app.tsx
```
This component (`JsonEditorWithMessageVariables`), which uses the `CodeEditor,` is exposed to be used by other plugins. It is being used only by the `stack_connectors` plugin which is responsible for all connectors across Kibana. The following connectors are using the `CodeEditor`:

- D3
- Index
- GenAI
- OpsGenie
- Tines
- Torq
- Webhook
- Cases webhook

From rules, only the Elasticsearch query rule is using the `CodeEditor`.

Some of the connectors are owned by ResponseOps, and some of them are by Security Solution. Also, the connector and rules flyout, which expose the connector/rule forms, which in turn expose the `CodeEditor,` can be used by any plugin. I know that the connector forms are being used by the Security solution and the rule and connector forms are being used by Observability and Discover. @elastic/response-ops Is there any other solution/plugin that uses our components?

kc13greiner committed:
Code Review from Patryk K. - removing TODO

cee-chen reviewed (approved):
Apologies for the late/blocking approval Kurt, I hadn't realized the EUI team was still listed as a codeowner of any Kibana files (rather than just being pinged for info). I'll defer to @rshen91's expertise on the touched kibana_react files, but in general what I saw looked good!

jeramysoucy commented on x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/rule_editor_panel/json_rule_editor.tsx:
```diff
@@ -97,7 +97,7 @@ export const JSONRuleEditor = (props: Props) => {
     >
       <Fragment>
-        <CodeEditorField
+        <CodeEditorField // TODO theme provided in context in x-pack/plugins/security/public/management/role_mappings/role_mappings_management_app.tsx
```
Found an issue here. When the space default is dark and the user setting is light, the JSON rule editor in Role Mappings is rendered dark.
<img width="1438" alt="Screenshot 2023-07-07 at 2 26 48 PM" src="https://github.com/elastic/kibana/assets/103939324/d59854e6-fd94-4594-8b6a-e266d29ea3e4">

kc13greiner replied:
Nice find! We will address this as discussed on Slack

jeramysoucy reviewed (dismissed):
Found just one issue in Kibana Security code. Our code editors look good!

kc13greiner committed:
Removing TODOs from reviewed CodeEditors

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner committed:
Fixing mergemain issue

cnasikas reviewed (approved):
ResponseOps changes LGTM! We rested with @XavierM and we found no issues. Solutions should verify their Connectors but we think it is safe to merge the PR.

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner commented:
@dasansol92 you were tagged as a reviewer, did you have any feedback?

patrykkopycinski commented:
@kc13greiner @dasansol92 is on PTO, feel free to merge it :)

ThomThomson commented on src/plugins/presentation_util/public/components/expression_input/expression_input.tsx:
```diff
@@ -75,7 +75,7 @@ export const ExpressionInput = (props: ExpressionInputProps) => {
   return (
     <div style={{ height, ...style }} {...{ rest }}>
-      <CodeEditor
+      <CodeEditor // TODO CodeOwners will need to help with this one
```
@kc13greiner this one is actually not used at the moment. It was likely meant to replace `x-pack/plugins/canvas/public/components/expression_input/expression_input.tsx` but never actually did so. I think it's okay to leave this one unverified for the moment.

kc13greiner replied:
Excellent! Thank you!

kc13greiner committed:
Removing todos

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

cnasikas commented on x-pack/plugins/stack_alerts/public/rule_types/es_query/expression/es_query_expression.tsx:
```diff
@@ -244,7 +244,7 @@ export const EsQueryExpression: React.FC<
         }
       >
-        <CodeEditor
+        <CodeEditor // TODO unable to find context. Need CodeOwners to help
```
nit: I think we can remove this.

kc13greiner replied:
Nice catch! Removed here - https://github.com/elastic/kibana/pull/159638/commits/bdbe9a83e97125ba271e026572a0c5818f56dcc2

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

kc13greiner committed:
Code Review feedback from Christos N. - removing TODO

kc13greiner committed:
Removing type error to avoid breaking changes

kc13greiner committed:
CodeReview feedback Oliver G. - Removing TODO

kc13greiner committed:
Merge branch 'main' into bugfix/provide_theme_in_context_for_useTheme_hook

ogupte reviewed (approved):
APM LGTM

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.9|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/161707">](https://github.com/elastic/kibana/pull/161707)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
